### PR TITLE
Add support for dynamic redirection of host/container ports

### PR DESCRIPTION
**Pull request by paulc**
Hi,

The attached pull-request adds support for dynamically configuring redirection between host/container ports.

This is one of the features I find most useful in Docker and I have previously used home-grown scripts to do this in Jails so would be great to support in Bastille.

This allows you to do things like:

```
# bastille rdr dev1 --tcp 2001 22
# bastille rdr dev1 --list
rdr on em0 inet proto tcp from any to any port = 2001 -> 10.17.89.1 port 22
# bastille rdr dev1 --udp 2053 53
# bastille rdr dev1 --list
rdr on em0 inet proto tcp from any to any port = 2001 -> 10.17.89.1 port 22
rdr on em0 inet proto udp from any to any port = 2053 -> 10.17.89.1 port 53
# bastille rdr dev1 --clear
nat cleared
```

To enable this you just need to add one `rdr-anchor` rule to `pf.conf`

```
ext_if="vtnet0"

set block-policy return
scrub in on $ext_if all fragment reassemble
set skip on lo

table <jails> persist
nat on $ext_if from <jails> to any -> ($ext_if)

## static rdr example
## rdr pass inet proto tcp from any to any port {80, 443} -> 10.17.89.45

# Enable dynamic rdr (see below)
rdr-anchor "rdr/*"

block in all
pass out quick modulate state
antispoof for $ext_if inet
pass in inet proto tcp from any to any port ssh flags S/SA keep state

# make sure you also open up ports that you are going to use for dynamic rdr
# pass in inet proto tcp from any to any port <rdr-start>:<rdr-end> flags S/SA keep state
# pass in inet proto udp from any to any port <rdr-start>:<rdr-end> flags S/SA keep state

```

### CODE REVIEW

**Comment by cedwards** [comment]
This is great! Thank you. I had something in mind just like this in my roadmap, but you beat me to it.

I also appreciate that you updated all relevant documentation. Bonus points to getting merged.

I have a few comments before we do that... I'll add some notes in the PR, but there's a couple typos and we have very few `-` or `--` options in Bastille now I think it'd fit more with the standard if we dropped those.

`bastille rdr TARGET tcp 2001 22`
`bastille rdr TARGET list`
...etc.

This would probably be trivial to add to the template system as well. Overall, I am looking forward to the addition.

File: README.md [hunk 1]
```diff
@@ -722,6 +749,28 @@ ishmael ~ # bastille cp ALL /tmp/resolv.conf-cf etc/resolv.conf
 /tmp/resolv.conf-cf -> /usr/local/bastille/jails/unbound0/root/etc/resolv.conf
 ```
 
+bastille-rdr
+------------
+
+`bastille rdr` allows yiou to configure dynamic rdr rules for your containers
```
Comment (cedwards) [inline README.md]: "yiou"

File: docs/chapters/networking.rst [hunk 2]
```diff
@@ -131,6 +138,23 @@ The `rdr pass ...` will redirect traffic from the host firewall on port X to
 the ip of Container Y. The example shown redirects web traffic (80 & 443) to the
 containers at `10.17.89.45`.
 
+  ## dynamic rdr anchor (see below)
+  rdr-anchor "rdr/*"
+
+The `rdr-anchor "rdr/*"` anables dynamic rdr rules to be setup using the
```
Comment (cedwards) [inline docs/chapters/networking.rst]: "anables"

File: docs/chapters/networking.rst [hunk 3]
```diff
@@ -131,6 +138,23 @@ The `rdr pass ...` will redirect traffic from the host firewall on port X to
...
+
+The `rdr-anchor "rdr/*"` anables dynamic rdr rules to be setup using the 
+`bastille rdr` command at runtime - eg.
+
+  bastille rdr <jail> --tcp 2001 22 # Redirects tcp port 2001 on host to 22 on jail
+  bastille rdr <jail> --udp 2053 53 # Same for udp
+  bastille rdr <jail> --list        # List dynamic rdr rules
+  bastille rdr <jail> --clear       # Clear dynamic rdr rules
+
+  Note that if you are rediirecting ports where the host is also listening
```
Comment (cedwards) [inline docs/chapters/networking.rst]: "rediirecting"

File: docs/chapters/subcommands/rdr.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
+===
+rdr
+===
+
+`bastille rdr` allows yiou to configure dynamic rdr rules for your containers
```
Comment (cedwards) [inline docs/chapters/subcommands/rdr.rst]: "yiou"

**Comment by paulc** [comment]
Hi,

I've updated the pull request to switch from `--option` to `option` and hopefully fix the typos (apologies for these - my typing isn't the best).

Paul

**Comment by cedwards** [comment]
Thank you. I think this is a very cool addition.

**Comment by hackacad** [comment]
What about updating the documentation as seen below. 

(1) This might make pf.conf easier to configure/understand. (config rdr_tcp/udp)
(2) Allow rdr ports only on jail IP addresses
(3) Restrict sshd to physical interface
(4) add labels for autoconfigured open ports (rdr_tcp/udp)

```
ext_if="ix1"
rdr_tcp = "{80, 443}" # bastille TCP ports
#rdr_udp = "{53}" # bastille UDP ports

set block-policy return
scrub in on $ext_if all fragment reassemble
set skip on lo

table <jails> persist
nat on $ext_if from <jails> to any -> ($ext_if)

## static rdr example
## rdr pass inet proto tcp from any to any port {80, 443} -> 10.17.89.45

# Enable dynamic rdr (see below)
rdr-anchor "rdr/*"

block in all
pass out quick modulate state
antispoof for $ext_if inet

pass in inet proto tcp from any to $ext_if port ssh flags S/SA keep state                                                                                                                    pass in inet proto tcp from any to <jails> port $rdr_tcp flags S/SA keep state label bastille_tcp_ports
#pass in inet proto udp from any to <jails> port $rdr_udp label bastille_udp_ports
```


Anyway, amazing feature!

**Pull request merged**